### ISSUE TITLE
Image shifted

### ISSUE THREAD

**Issue by andreadompe** [bpampuch/pdfmake#355]
Hi all!
I've already red other issues but i didn't find a solution,
this is my problem:
i've copied the playground images example and after the PDF creation on the browser everything is ok but if i open the pdf file all the images are shifted to the right...

I've already  tried with jpg, png but the result is the same, also switching from Windows 7 to iOS.

How can i fix this?

Greeting

Andrea

**Comment by andreadompe** [comment]
No one has the same problem???

**Comment by gabucito** [comment]
I just tested and I have the same problem as you.  All the images get shifted to the right when you reopen the pdfs.

**Comment by gabucito** [comment]
I took a quick look at the files and it seems that this line in elementWriter.js is what is causing the problem:
<pre>image.x = context.x + (image.x || 0);</pre>

I am taking a guess but it seems the value for image.x do not get resetted when you reopen the pdf which causes the problem.

If you are not setting an initial position then changing the above line to: 
<pre>image.x = context.x;</pre>
might help.

This might break your initial positioning, if you set one, for the images so beware.  Hopefully one of the devs will take a look into this soon.

**Comment by andreadompe** [comment]
Now it works perfectly!
Thanks gabucito!

**Comment by EpicWolverine** [comment]
Another solution I've found is to redefine your docDefinition every time you generate the PDF. In other words, structure it like
```
var savePDF = function(){
    var docDefinition = {...};
    pdfMake.createPdf(docDefinition);
}
```
and call `savePDF()` every time. @Elyahou has a cleaner solution at [#376](https://github.com/bpampuch/pdfmake/issues/376#issuecomment-137985898) but I figured this may help someone.

By the way, this issue has been duplicated multiple times: #389, #446, #592
